torch/profiler/profiler.py

```diff
@@ -878,13 +884,50 @@ def __init__(self) -> None:
         """
         self._registered = False
         self._execution_trace_running = False
+        self.extra_resources_collection = False
+        self.resources_dir: str = ""
 
     def __del__(self):
         """
         Calls unregister_callback() to make sure to finalize outputs.
         """
         self.unregister_callback()
 
+    @staticmethod
+    def build_execution_trace_obs_from_env() -> Optional["ExecutionTraceObserver"]:
+        """
+        Returns an ExecutionTraceObserver instance if the environment variable
+        ENABLE_PYTORCH_EXECUTION_TRACE is set to 1, otherwise returns None.
+        """
+        if os.environ.get("ENABLE_PYTORCH_EXECUTION_TRACE", "0") == "1":
+            fp = tempfile.NamedTemporaryFile("w+t", suffix=".et.json", delete=False)
+            fp.close()
+            et = ExecutionTraceObserver()
+            et.register_callback(fp.name)
+            et.set_extra_resource_collection()
```

Comment:
It seems much cleaner to call set_extra_resource_collection(True) instead of use default value "False" and check the en. variable to overwrite it inside set_extra_resource_collection. We can remove the logic to check env. variable inside set_extra_resource_collection.